src/modules/roc_core/format_tid.h

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2019 Roc authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/format_tid.h
+//! @brief Retrieve the current thread.
+
+#ifndef ROC_CORE_FORMAT_TID_H_
+#define ROC_CORE_FORMAT_TID_H_
+
+#include <sys/types.h>
+
+namespace roc {
+namespace core {
+
+//! Retrieve and format current time.
+//
+//!
+//! @returns
+//!  the thread id as an int
+//!
+//! @note
+//!  This function should not log anything because it is used
+//!  in the logger implementation.
+int format_tid();
```

Comment:
I suggest to convert the signature to the following:

```
bool format_tid(char* buf, size_t bufsz);
```

(similar to format_time)

This way we'll avoid using platform-dependent types (pid_t/int in this case) in the declaration of platform-independent function. The function will gather thread ID and format it as string, and it will not matter what specific types it uses.